src/custom-elements/src/components/text-input/text-input.tsx

```diff
@@ -0,0 +1,149 @@
+import { Component, Element, Method, Prop, State, Watch, h } from '@stencil/core';
+import { CHI_STATES, ChiStates } from '../../constants/states';
+import { ICON_COLORS, IconColors } from '../../constants/color';
+import { TEXT_INPUT_SIZES, TextInputSizes } from '../../constants/size';
+import { TEXT_INPUT_TYPES, TextInputTypes } from '../../constants/constants';
+
+@Component({
+  tag: 'chi-text-input',
+  styleUrl: 'text-input.scss',
+  scoped: true
+})
+
+export class TextInput {
+  @Element() el: HTMLInputElement;
+  /**
+   * To define type of Text input
+   */
+  @Prop() type: TextInputTypes = 'text';
+  /**
+   * To define state color of Text input
+   */
+  @Prop({ reflect: true }) state: ChiStates;
+  /**
+   * To add a left positioned icon
+   */
+  @Prop({ reflect: true }) iconLeft: string;
+  /**
+   * To add a right positioned icon
+   */
+  @Prop({ reflect: true }) iconRight: string;
+  /**
+   * To define color of left icon
+   */
+  @Prop({ reflect: true }) iconLeftColor: IconColors;
+  /**
+   * To define color of right icon
+   */
+  @Prop({ reflect: true }) iconRightColor: IconColors;
+  /**
+   * To define size of Text input
+   */
+  @Prop({ reflect: true }) size ?: TextInputSizes = 'md';
+  /**
+   * To define placeholder of Text input
+   */
+  @Prop({ reflect: true }) placeholder: string;
+  /**
+   * To define value of Text input
+   */
+  @Prop({ reflect: true }) value: string;
+  /**
+   * To disable Text input
+   */
+  @Prop({ reflect: true }) disabled = false;
+
+  @State() status: string;
+
+  @Watch('state')
+  stateValidation(newValue: ChiStates) {
+    const validValues = CHI_STATES.join(', ');
+
+    if (newValue && !CHI_STATES.includes(newValue)) {
+      throw new Error(`${newValue} is not a valid state for input. If provided, valid values are: ${validValues}. `);
+    }
+  }
+
+  @Watch('size')
+  sizeValidation(newValue: TextInputSizes) {
+    const validValues = TEXT_INPUT_SIZES.join(', ');
+
+    if (newValue && !TEXT_INPUT_SIZES.includes(newValue)) {
+      throw new Error(`${newValue} is not a valid size for input. If provided, valid values are: ${validValues}. `);
+    }
+  }
+
+  colorValidation(newValue: IconColors) {
```

Comment:
@janibolkvadze Did you forget to include the Watcher here?

Reply:
No, this is a method called by 2 watchers:
```
@Watch('iconLeftColor')
  iconLeftColorValidation(newValue: IconColors) {
    this.colorValidation(newValue);
  }

@Watch('iconRightColor')
iconRightColorValidation(newValue: IconColors) {
  this.colorValidation(newValue);
}
```